homeassistant/components/climate/ecobee.py

```diff
@@ -246,25 +264,38 @@ def turn_away_mode_off(self):
 
     def set_temperature(self, **kwargs):
         """Set new target temperature."""
-        if kwargs.get(ATTR_TARGET_TEMP_LOW) is not None and \
-           kwargs.get(ATTR_TARGET_TEMP_HIGH) is not None:
-            high_temp = int(kwargs.get(ATTR_TARGET_TEMP_LOW))
-            low_temp = int(kwargs.get(ATTR_TARGET_TEMP_HIGH))
+        low_temp = None
+        high_temp = None
+        temp = None
+
+        if kwargs.get(ATTR_TARGET_TEMP_LOW) is not None:
+            low_temp = int(kwargs.get(ATTR_TARGET_TEMP_LOW))
+
+        if kwargs.get(ATTR_TARGET_TEMP_HIGH) is not None:
+            high_temp = int(kwargs.get(ATTR_TARGET_TEMP_HIGH))
+
+        if kwargs.get("temperature") is not None:
+            temp = int(kwargs.get("temperature"))
+
+        if self.current_operation == STATE_HEAT:
+            low_temp = temp
+            high_temp = temp + 20
+        elif self.current_operation == STATE_COOL:
+            low_temp = temp - 20
+            high_temp = temp
```

Comment:
I am not sure why you added the condition on self.hold_temp. I know there are several in the code, but I think these are misguided. I added in my work on the generalized hold feature a lookup to check what the thermostat has as setting for hold (there are various options, 'nextTransition', 'indefinite', 2 hours, 4 hours, some specific time, etc.). We should not set an indefinite hold when there already is a hold. We should set an indefinite hold when the user preference is such. 

However, I am ok with sticking this there for now, just to be consistent with the other places where this happens (albeit it should all be corrected).